### PR TITLE
Helper method for ros param, set cmd_vel timeout threshold with ros param.

### PR DESCRIPTION
**Pull request by markwomack**
This is a first pass. Ros params is going to need more work in order to dynamically change values (Issue #20)

### CODE REVIEW

File: Arduino/victoria_teensy_ros_node/victoria_teensy_ros_node.ino [hunk 1]
```diff
@@ -84,10 +85,13 @@ bool mag_err = false;
...
 ros::NodeHandle ros_nh;
 
+// Parameter helper
+RosParamHelper ros_param_helper(ros_nh);
+
 // ROS cmd_vel subscriber
 // Threshold, in seconds, to wait for a cmd_vel message else take action
 // to stop robot activity
-const double CMD_VEL_TIMEOUT_THRESHOLD(0.2); // 200 milliseconds
+double CMD_VEL_TIMEOUT_THRESHOLD;
```
Comment (griswaldbrooks) [inline Arduino/victoria_teensy_ros_node/victoria_teensy_ros_node.ino]: non-const parameters should be lower case.

File: Arduino/victoria_teensy_ros_node/RosParamHelper.h [hunk 2]
```diff
@@ -0,0 +1,80 @@
+#ifndef ROS_PARAM_HELPER_h
```
Comment (griswaldbrooks) [inline Arduino/victoria_teensy_ros_node/RosParamHelper.h]: Please refer to the Google Style Guide https://google.github.io/styleguide/cppguide.html#The__define_Guard
(But the ROS style guide first which is refers to Google on this issue http://wiki.ros.org/CppStyleGuide)

Let's say it should be
`#ifndef VICTORIA_TEENSY_ROS_NODE_ROS_PARAM_HELPER_H_`

I think that would be right.

File: Arduino/victoria_teensy_ros_node/RosParamHelper.h [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+#define ROS_PARAM_HELPER_h
+
+#include <ros.h>
+
+class RosParamHelper {
```
Comment (griswaldbrooks) [inline Arduino/victoria_teensy_ros_node/RosParamHelper.h]: All classes and methods should have doxygen comments.

File: Arduino/victoria_teensy_ros_node/RosParamHelper.h [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+public:
+  RosParamHelper(ros::NodeHandle nh);
+  
+  /**
+   * Retrieves int parameter from ros and sets it
```
Comment (griswaldbrooks) [inline Arduino/victoria_teensy_ros_node/RosParamHelper.h]: In this case the doxygen should start with `\brief`

File: Arduino/victoria_teensy_ros_node/RosParamHelper.h [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+   * into the param.
+   *
```
Comment (griswaldbrooks) [inline Arduino/victoria_teensy_ros_node/RosParamHelper.h]: `\param` should be listed when specifying input params.
`\param[out]` for output parameters.

File: Arduino/victoria_teensy_ros_node/RosParamHelper.h [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+class RosParamHelper {
+
+public:
+  RosParamHelper(ros::NodeHandle nh);
+  
+  /**
+   * Retrieves int parameter from ros and sets it
+   * into the param.
+   * 
+   * Returns true if parameter was found in ros.
```
Comment (griswaldbrooks) [inline Arduino/victoria_teensy_ros_node/RosParamHelper.h]: `\return` for return values.

File: Arduino/victoria_teensy_ros_node/RosParamHelper.cpp [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+bool RosParamHelper::getParam(const char* name, int* param) {
+  return nh_.getParam(name, param);
+}
+
+bool RosParamHelper::getParam(const char* name, int* param, int defaultValue) {
+  if (nh_.getParam(name, param)) {
+    return true;
+  }
+  param[0] = defaultValue;
```
Comment (griswaldbrooks) [inline Arduino/victoria_teensy_ros_node/RosParamHelper.cpp]: should be `(*param) = defaultValue`

File: Arduino/victoria_teensy_ros_node/RosParamHelper.cpp [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+bool RosParamHelper::getParam(const char* name, float* param) {
+  return nh_.getParam(name, param);
+}
+
+bool RosParamHelper::getParam(const char* name, float* param, float defaultValue) {
+  if (nh_.getParam(name, param)) {
+    return true;
+  }
+  param[0] = defaultValue;
```
Comment (griswaldbrooks) [inline Arduino/victoria_teensy_ros_node/RosParamHelper.cpp]: As above.
should be `(*param) = defaultValue`

File: Arduino/victoria_teensy_ros_node/RosParamHelper.cpp [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  return false;
+}
+
+int RosParamHelper::getParam(const char* name, int defaultValue) {
+  int paramValue;
+  getParam(name, &paramValue, defaultValue);
+  return paramValue;
+}
+
+bool RosParamHelper::getParam(const char* name, float* param) {
```
Comment (griswaldbrooks) [inline Arduino/victoria_teensy_ros_node/RosParamHelper.cpp:25]: This should probably be templated. Let's make that a ticket.

File: Arduino/victoria_teensy_ros_node/RosParamHelper.cpp [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+bool RosParamHelper::getParam(const char* name, float* param, float defaultValue) {
+  if (nh_.getParam(name, param)) {
+    return true;
+  }
+  param[0] = defaultValue;
+  return false;
+}
+
+float RosParamHelper::getParam(const char* name, float defaultValue) {
+  int paramValue;
```
Comment (griswaldbrooks) [inline Arduino/victoria_teensy_ros_node/RosParamHelper.cpp]: `float`

**Review by griswaldbrooks — CHANGES_REQUESTED** [review]

File: Arduino/victoria_teensy_ros_node/RosParamHelper.h (hunk 2 above)
Comment (markwomack) [reply Arduino/victoria_teensy_ros_node/RosParamHelper.h]: Done

File: Arduino/victoria_teensy_ros_node/RosParamHelper.h (hunk 3 above)
Comment (markwomack) [reply Arduino/victoria_teensy_ros_node/RosParamHelper.h]: done.

File: Arduino/victoria_teensy_ros_node/RosParamHelper.h (hunk 4 above)
Comment (markwomack) [reply Arduino/victoria_teensy_ros_node/RosParamHelper.h]: done

File: Arduino/victoria_teensy_ros_node/RosParamHelper.h (hunk 5 above)
Comment (markwomack) [reply Arduino/victoria_teensy_ros_node/RosParamHelper.h]: done

File: Arduino/victoria_teensy_ros_node/RosParamHelper.h (hunk 6 above)
Comment (markwomack) [reply Arduino/victoria_teensy_ros_node/RosParamHelper.h]: done

File: Arduino/victoria_teensy_ros_node/RosParamHelper.cpp (hunk 7 above)
Comment (markwomack) [reply Arduino/victoria_teensy_ros_node/RosParamHelper.cpp]: done

File: Arduino/victoria_teensy_ros_node/RosParamHelper.cpp (hunk 8 above)
Comment (markwomack) [reply Arduino/victoria_teensy_ros_node/RosParamHelper.cpp]: done

File: Arduino/victoria_teensy_ros_node/RosParamHelper.cpp (hunk 10 above)
Comment (markwomack) [reply Arduino/victoria_teensy_ros_node/RosParamHelper.cpp]: done

File: Arduino/victoria_teensy_ros_node/RosParamHelper.cpp (hunk 9 above)
Comment (markwomack) [reply Arduino/victoria_teensy_ros_node/RosParamHelper.cpp:25]: #27

**Review by griswaldbrooks — APPROVED** [review]

**Pull request merged**